frontend/src/components/Agreements/AgreementDetails/AgreementHistoryPanel.jsx

```diff
@@ -0,0 +1,173 @@
+        return `${classAndTitle} updated by ${user_name}.`;
+    } else if (historyItem.event_type === "DELETED") {
+        return `${classAndTitle} deleted by ${user_name}.`;
+    }
+    return `${className} ${historyItem.event_type} ${user_name}`;
+};
+
+const getPropertyLabel = (className, fieldName) => {
+    if (className === "BudgetLineItem") return convertCodeForDisplay("budgetLineItemPropertyLabels", fieldName);
+    return convertCodeForDisplay("agreementPropertyLabels", fieldName);
+};
+
+const usersToNames = (users) => {
+    return users.map((user) => user.full_name);
+};
+
+const ChangesDetails = ({ historyItem }) => {
+    const changes = historyItem.changes;
+    const eventType = historyItem.event_type;
+    if (eventType != "UPDATED") return;
+    console.log("ChangesDetails.changes:", changes);
+    const prepChanges = Object.entries(changes).map(([key, change]) => {
+        if ("collection_of" in change) {
+            const added = change.collection_of == "User" ? usersToNames(change.added) : change.added;
+            const deleted = change.collection_of == "User" ? usersToNames(change.deleted) : change.deleted;
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, key),
+                isCollection: true,
+                added: added,
+                deleted: deleted,
+            };
+        } else if (key === "procurement_shop_id") {
+            const new_val = historyItem.event_details?.procurement_shop?.name;
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, "procurement_shop"),
+                to: new_val,
+            };
+        } else if (key === "product_service_code_id") {
+            const new_val = historyItem.event_details?.product_service_code?.name;
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, "product_service_code"),
+                to: new_val,
+            };
+        } else if (key === "project_officer") {
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, "project_officer"),
+            };
+        } else if (key === "research_project_id") {
+            const new_val = historyItem.event_details?.research_project?.title;
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, "research_project"),
+                to: new_val,
+            };
+        } else if (key === "can_id") {
+            const new_val = historyItem.event_details?.can?.number;
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, "can"),
+                to: new_val,
+            };
+        } else
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, key),
+                from: change.old,
+                to: change.new,
+            };
+    });
+
+    return (
+        <dl>
+            {prepChanges.map((change) => (
```

Comment:
`prepChanges` is a little ambiguous...maybe a rename to something a little clearer like `logList`?

Reply:
I made a slight clarification.  I didn't go with logList since to me that's sounds like list of history records rather than the field-level changes in a single history record.